src/server/v2.0/handler/project.go

```diff
@@ -2,33 +2,211 @@ package handler
+		log.Errorf("failed to determine whether only admin can create projects: %v", err)
+		return a.SendError(ctx, fmt.Errorf("failed to determine whether only admin can create projects: %v", err))
+	}
+
+	secCtx, _ := security.FromContext(ctx)
+	if onlyAdmin && !(secCtx.IsSysAdmin() || secCtx.IsSolutionUser()) {
+		log.Errorf("Only sys admin can create project")
+		return a.SendError(ctx, errors.ForbiddenError(nil).WithMessage("Only system admin can create project"))
+	}
+
+	req := params.Project
+
+	if req.RegistryID != nil && !secCtx.IsSysAdmin() {
+		// only system admin can create the proxy cache project
+		return a.SendError(ctx, errors.ForbiddenError(nil).WithMessage("Only system admin can create proxy cache project"))
+	}
+
+	// populate storage limit
+	if config.QuotaPerProjectEnable() {
+		if req.StorageLimit == nil || *req.StorageLimit == 0 || !secCtx.IsSysAdmin() {
+			setting, err := config.QuotaSetting()
+			if err != nil {
+				log.Errorf("failed to get quota setting: %v", err)
+				return a.SendError(ctx, fmt.Errorf("failed to get quota setting: %v", err))
+			}
+			defaultStorageLimit := setting.StoragePerProject
+			req.StorageLimit = &defaultStorageLimit
+		}
+	} else {
+		// ignore storage limit when quota per project disabled
+		req.StorageLimit = nil
+	}
+
+	if req.Metadata == nil {
+		req.Metadata = &models.ProjectMetadata{}
+	}
+
+	// accept the "public" property to make replication work well with old versions(<=1.2.0)
+	if req.Public != nil && req.Metadata.Public == "" {
+		req.Metadata.Public = strconv.FormatBool(*req.Public)
+	}
+
+	// populate public metadata as false if it isn't set
+	if req.Metadata.Public == "" {
+		req.Metadata.Public = strconv.FormatBool(false)
+	}
+
+	if err := a.validateProjectReq(ctx, req); err != nil {
+		return a.SendError(ctx, err)
+	}
+
+	var ownerID int
+	// set the owner as the system admin when the API being called by replication
+	// it's a solution to workaround the restriction of project creation API:
+	// only normal users can create projects
+	if secCtx.IsSolutionUser() {
+		ownerID = 1
+	} else {
+		ownerName := secCtx.GetUsername()
+		user, err := a.userMgr.GetByName(ctx, ownerName)
+		if err != nil {
+			return a.SendError(ctx, err)
+		}
+		ownerID = user.UserID
+	}
+
+	p := &project.Project{
+		Name:       req.ProjectName,
+		OwnerID:    ownerID,
+		RegistryID: lib.Int64Value(req.RegistryID),
+	}
+	lib.JSONCopy(&p.Metadata, req.Metadata)
+
+	projectID, err := a.projectCtl.Create(ctx, p)
+	if err != nil {
+		return a.SendError(ctx, err)
+	}
+
+	if req.StorageLimit != nil {
+		referenceID := quota.ReferenceID(projectID)
+		hardLimits := types.ResourceList{types.ResourceStorage: *req.StorageLimit}
+		if _, err := a.quotaCtl.Create(ctx, quota.ProjectReference, referenceID, hardLimits); err != nil {
+			return a.SendError(ctx, fmt.Errorf("failed to create quota for project: %v", err))
+		}
+	}
+
+	// create a default retention policy for proxy project
+	if req.RegistryID != nil {
```

Comment:
I have filed an issue for it, whether to create a default policy, it can be discussed.